spec/models/grda_warehouse/cohort_spec.rb

```diff
@@ -134,4 +146,49 @@
       expect(updated_collection.name).to eq('New Name')
     end
   end
+
+  describe 'column_state handling' do
+    let!(:test_column) { build :user_string_cohort_column_1 }
+
+    before do
+      # Add test column to cohort's column_state
+      columns = cohort.column_state || []
+      test_column.cohort_column.activate
+      columns << test_column
+      cohort.update(column_state: columns)
+    end
+
+    it 'removes inactive columns from column_state' do
+      expect(cohort.column_state.map(&:class_name)).to include('CohortColumns::UserString1')
```

Comment:
These tests would have to be updated to something like:
```suggestion
      expect(cohort.column_state.map(&:class_name)).to contain_exactly(@all_columns)
```